docs/source/notes/run_name.md

```diff
@@ -0,0 +1,91 @@
+# 🏃‍♀️ Run Name
+
+The `run_name` is a string used to name a specific training run. Naming your training run has many benefits and your `run_name` will show up in many places as you utilize Composer.
+
+## Run Name Creation
+
+The `run_name` argument is an optional argument to the  {class}`~.Trainer`. 
+
+There are two ways to get a `run_name`.
+
+You create your own `run_name` and pass it to the trainer, like so:
+ 
+<!--pytest-codeblocks:skip-->
+```python
+from torchvision import datasets, transforms
+from torch.utils.data import DataLoader
+
+from composer import Trainer
+from composer.models import mnist_model
+
+transform = transforms.Compose([transforms.ToTensor()])
+dataset = datasets.MNIST("data", train=True, download=True, transform=transform)
+train_dataloader = DataLoader(dataset, batch_size=128)
+run_name = 'my-cool-run-name'
+
+trainer = Trainer(
+    model=mnist_model(num_classes=10),
+    train_dataloader=train_dataloader,
+    max_duration="2ep",
+        run_name=run_name,
+)
+trainer.fit()
+```
+        
+You can instead let the trainer create a `run_name` for you. The one created for you is the timestamp followed by a [coolname](https://github.com/alexanderlukanin13/coolname); e.g. `‘1657932618-infrared-ferret’`
+
+
+## How the Run Name is Used
+
+This run_name will be added as an attribute to {class}`~.State` and it is used by various other pieces of the composer infrastructure as described below.
+
+The run_name is often used  in the Composer as placeholder in a format string. This means that if a string is specified to name a file, like `'{run_name}-foo-bar'` for example, then that placeholder will get filled in by the actual `run_name` at runtime, so the file will actually be named `‘1657932618-infrared-ferret-foo-bar’`.
+
+### Run Names in Checkpoint Saving
```

Comment:
This is a really great list, but I'm a bit concerned it will be difficult to keep this section up to date. I think the docstrings in each of these sections explain how the run name will be used. Perhaps would it be possible to link to a _few_ places where the run name is used, and then add back links in _all_ places where it is used to this document?